src/components/common/Switch.js

```diff
@@ -9,41 +9,34 @@ import "styles/Switch.scss";
  * @param {JSX} onImg (optional): (JSX) element to be displayed on switch body when set to "on"
  * @param {JSX} offImg (optional): (JSX) element to be displayed on switch body when set to "off"
  */
-export default class Switch extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      on: !this.props.on ? false : this.props.on,
-    };
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.props.on !== prevProps.on) {
-      this.setState({
-        on: this.props.on,
-      });
+
+const Switch = (props) => {
+  const [state, setState] = useState({
```

Comment:
This works, but does _on_ need to be nested inside a state object?

Reply:
Well, not at all but i intend to follow state and setState naming convention from class to functional component and it's very clear for someone who hasn't use hook, considering that we are actually using the same state variable name throughout the component. Plus there is no any side effect of using that.
**But, if required, i will change it**

Reply:
This is unlikely to matter for this particular component, but in general I think separating state is better than a catch-all state